lxd/storage/drivers/driver_lvm.go

```diff
@@ -526,6 +528,61 @@ func (d *lvm) Update(changedConfig map[string]string) error {
 		d.logger.Debug("Thin pool volume renamed", logger.Ctx{"vg_name": d.config["lvm.vg_name"], "thinpool": d.thinpoolName(), "new_thinpool": changedConfig["lvm.thinpool_name"]})
 	}
 
+	size, ok := changedConfig["size"]
+	if ok {
+		sizeBytes, err := d.roundedSizeBytesString(size)
+		if err != nil {
+			return err
+		}
+
+		// Figure out loop path
+		loopPath := loopFilePath(d.name)
+
+		if filepath.IsAbs(d.config["source"]) {
+			loopPath = d.config["source"]
+		}
+
+		// Resize loop file
+		f, err := os.OpenFile(loopPath, os.O_RDWR, 0600)
+		if err != nil {
+			return err
+		}
+
+		err = unix.Fallocate(int(f.Fd()), 0, 0, sizeBytes)
```

Comment:
@monstermunchkin im not sure if its the same, but for resizing block volume files for VMs on dir pools, we use `f.Truncate()` here, so if that does the trick it would be good to use the same approach for resizing block files throughout the code base.

https://github.com/lxc/lxd/blob/fef27435fecbcf686744c6f6de99d3ead79cfb34/lxd/storage/drivers/utils.go#L303-L310

Reply:
It seems the same. I've updated the storage code to use `f.Truncate()` instead.